contrib/drivers/ceph/ceph_metrics.go

```diff
@@ -0,0 +1,192 @@
+	}
+	metricToUnitMap := make(map[string]string)
+	for _, resources := range configs.Cfgs {
+		switch resources.Resource {
+		//ToDo: Other Cases needs to be added
+		case "pool":
+			for index, metricName := range resources.Metrics {
+
+				metricToUnitMap[metricName] = resources.Units[index]
+
+			}
+		}
+	}
+	return metricToUnitMap
+}
+
+// 	ValidateMetricsSupportList:- is  to check whether the posted metric list is in the uspport list of this driver
+// 	metricList-> Posted metric list
+//	supportedMetrics -> list of supported metrics
+func (d *MetricDriver) ValidateMetricsSupportList(metricList []string, resourceType string) (supportedMetrics []string, err error) {
+	var configs Configs
+
+	//Read supported metric list from yaml config
+	//Todo: Move this to read from file
+	source := []byte(data)
+	error := yaml.Unmarshal(source, &configs)
+	if error != nil {
+		log.Fatalf("Unmarshal error: %v", error)
+	}
+
+	for _, resources := range configs.Cfgs {
+		switch resources.Resource {
+		//ToDo: Other Cases needs to be added
+		case "pool":
+			for _, metricName := range metricList {
+				if metricInMetrics(metricName, resources.Metrics) {
+					supportedMetrics = append(supportedMetrics, metricName)
+
+				} else {
+					log.Infof("metric:%s is not in the supported list", metricName)
+				}
+			}
+		}
+	}
+	return supportedMetrics, nil
+}
+
+//	CollectMetrics: Driver entry point to collect metrics. This will be invoked by the dock
+//	metricsList-> posted metric list
+//	instanceID -> posted instanceID
+//	metricArray	-> the array of metrics to be returned
+func (d *MetricDriver) CollectMetrics(metricsList []string, instanceID string) ([]*model.MetricSpec, error) {
+
+	// get Metrics to unit map
+	metricToUnitMap := getMetricToUnitMap()
+	//validate metric support list
+	supportedMetrics, err := d.ValidateMetricsSupportList(metricsList, "pool")
+	if supportedMetrics == nil {
+		log.Infof("No metrics found in the  supported metric list")
+	}
+	metricMap, err := CollectMetrics(supportedMetrics, instanceID)
+
+	var tempMetricArray []*model.MetricSpec
+	for label_val, _ := range metricMap {
+		for _, element := range supportedMetrics {
+			val, _ := strconv.ParseFloat(metricMap[label_val][element], 64)
+			//Todo: See if association  is required here, resource discovery could fill this information
+			associatorMap := make(map[string]string)
+			associatorMap["cluster"] = "ceph"
+			associatorMap["pool"] = label_val
+			metricValue := &model.Metric{
+				Timestamp: getCurrentUnixTimestamp(),
+				Value:     val,
+			}
+			metricValues := make([]*model.Metric, 0)
+			metricValues = append(metricValues, metricValue)
+
+			metric := &model.MetricSpec{
+				InstanceID:   instanceID,
+				InstanceName: "",
+				Job:          "OpenSDS",
+				Labels:       associatorMap,
+				//Todo Take Componet from Post call, as of now it is only for volume
+				Component: "Pool",
+				Name:      fmt.Sprintf("%s_%s", associatorMap["cluster"], element),
```

Comment:
Name is just name of the metric in metric spec, not ceph exporter  name .
for example if ceph exporter  name is 'pool_read_bytes_total' , then corresponding metric spec field would be
componet=pool
name=read
unit=bytes
AggrType=Total